cmake/tests/cxx14_auto.cpp

```diff
@@ -0,0 +1,20 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2009 Andrey Semashev
+//  Copyright (c) 2015 Agustin Berge
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+template<typename T>
+auto check_f(T& i)
+{
+    T j = 1 + j;
+    return j;
+}
```

Comment:
At CMake time we don't know if the user is going to use nvcc for their code.

I can drop the additional (misleading) CXX14 CMake test and then users can manually disable CXX11 auto support. Would that be better?